src/main/java/org/opensrp/web/utils/InitialFormConfigUploadUtil.java

```diff
@@ -0,0 +1,392 @@
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import org.json.JSONObject;
+import org.opensrp.service.ClientFormService;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class InitialFormConfigUploadUtil {
+
+	private ArrayList<String> jsonPathForSubFormReferences = new ArrayList<>();
+	private ArrayList<String> jsonPathForRuleReferences = new ArrayList<>();
+	private ArrayList<String> jsonPathForPropertyFileReferences = new ArrayList<>();
+
+
+	public static void main(String[] mainArgs) throws IOException {
+		InitialFormConfigUploadUtil clientFormValidator = new InitialFormConfigUploadUtil();
+		clientFormValidator.performUpload();
+	}
+
+	public InitialFormConfigUploadUtil() {
+		ClientFormValidator clientFormValidator = new ClientFormValidator(new ClientFormService());
+		jsonPathForPropertyFileReferences = clientFormValidator.getJsonPathForPropertyFileReferences();
+		jsonPathForRuleReferences = clientFormValidator.getJsonPathForRuleReferences();
+		jsonPathForSubFormReferences = clientFormValidator.getJsonPathForSubFormReferences();
+	}
+
+	private Call generateHttpCall(String clientFormUrl, String basicAuthValue, RequestBody requestBody) {
+		Request request = new Request.Builder()
+				.header("Authorization", basicAuthValue)
+				.url(clientFormUrl)
+				.post(requestBody)
+				.build();
+		OkHttpClient httpClient = new OkHttpClient();
+		httpClient.setConnectTimeout(1, TimeUnit.SECONDS);
+		httpClient.setReadTimeout(1, TimeUnit.SECONDS);
+
+		return httpClient
+				.newCall(request);
+	}
+
+	private RequestBody generateRequestBody(String formVersion, String fileName, String fileContentType, File ymlFile,
+			String formRelation, String title) throws IOException {
+		MultipartBuilder multipartBuilder = new MultipartBuilder()
+				.type(MultipartBuilder.FORM)
+				.addFormDataPart("form_version", formVersion)
+				.addFormDataPart("form_identifier", fileName)
+				.addFormDataPart("form_name", title)
+				.addFormDataPart("form", fileName, RequestBody
+						.create(MediaType.parse(fileContentType), Files.readAllBytes(ymlFile.toPath())));
+
+		//.addPart(RequestBody.create(MediaType.parse("application/x-yaml"), Files.readAllBytes(ymlFile.toPath())));
+
+		if (formRelation != null) {
+			multipartBuilder.addFormDataPart("form_relation", formRelation);
+		}
+
+		return multipartBuilder.build();
+	}
+
+
+	public void performUpload() throws IOException {
+		/*
+		********************************
+		 THESE VALUES SHOULD BE CHANGED
+		********************************
+		*/
+
+		// Path to client project assets folder with json.form
+		String assetsFilePath = "/home/ona-kigamba/Documents/Projects/OpenSRP/opensrp-client-reveal/opensrp-reveal/src/main/assets/";
```

Comment:
Please make this value configurable

Reply:
What do you mean by making this configurable? I can move them to the main method and retain the comments. I believe the comments make it clear that the variable values can be changed. When the class is openned in Intellij IDEA, it will have the green play button next to the main method to indicate it can be run just by clicking the button.

Reply:
Can't we have them on a properties file and just have all the variables that need to change there?

Reply:
Yea, we definitely can. I guess the aim is to make it very simple to use. Have everything in one place so that it's as straight forward as possible. In case you still want this moved to a separate property file, we can still go ahead and move it outside